src/main/java/net/sf/jabref/exporter/SaveDatabaseAction.java

```diff
@@ -358,25 +358,23 @@ public void saveAs() throws Throwable {
             }
         }
 
-        if (f != null) {
-            File oldFile = panel.getBibDatabaseContext().getDatabaseFile();
-            panel.getBibDatabaseContext().setDatabaseFile(f);
-            Globals.prefs.put(JabRefPreferences.WORKING_DIRECTORY, f.getParent());
-            runCommand();
-            // If the operation failed, revert the file field and return:
-            if (!success) {
-                panel.getBibDatabaseContext().setDatabaseFile(oldFile);
-                return;
-            }
-            // Register so we get notifications about outside changes to the file.
-            try {
-                panel.setFileMonitorHandle(Globals.fileUpdateMonitor.addUpdateListener(panel,
-                        panel.getBibDatabaseContext().getDatabaseFile()));
-            } catch (IOException ex) {
-                LOGGER.error("Problem registering file change notifications", ex);
-            }
-            frame.getFileHistory().newFile(panel.getBibDatabaseContext().getDatabaseFile().getPath());
+        File oldFile = panel.getBibDatabaseContext().getDatabaseFile();
```

Comment:
I just saw there is a throwable in thrown in saveAs() (some lines above, can't mark it here directly)
Can you please check if this is necessary or if it can be replaced by some more specific exception type?